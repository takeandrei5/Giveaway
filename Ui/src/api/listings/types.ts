import { PaginatedResult } from '../../utils/types';

export type ItemData = { id: string; image: string; title: string; createdAt: Date };

export type CreateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export type UpdateListingRequest = {
	title: string;
	description: string;
	category: number;
	images: string[];
};

export interface FetchListingsResponse {
	listings: PaginatedResult<ItemData>;
}

export type ListingInformation = {
	id: string;
	title: string;
	description: string;
	category: number;
	createdAt: Date;
	images: string[];
};

export type OwnerInformation = {
	email: string;
	name: string;
	image: string;
};

export type FetchListingDetailsResponse = { listingInfo: ListingInformation } & { ownerInfo: OwnerInformation };
