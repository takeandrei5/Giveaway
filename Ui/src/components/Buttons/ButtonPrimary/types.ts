import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = Omit<ButtonProps, 'color'>;
