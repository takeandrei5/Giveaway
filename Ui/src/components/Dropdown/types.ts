export type DropdownProps = {
	id: string;
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
	name: string;
	value: string | number | readonly string[];
	isInvalid?: false;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
