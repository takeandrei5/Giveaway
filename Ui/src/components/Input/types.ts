export type InputProps = {
	id: string;
	name: string;
	placeholder: string;
	value?: string;
	onBlur?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onChange?: (e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) => void;
	onKeyDown?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyPress?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	onKeyUp?: (e: React.KeyboardEvent<HTMLInputElement> | React.KeyboardEvent<HTMLTextAreaElement>) => void;
	disabled?: boolean;
	multiline?: boolean;
	isInvalid?: boolean;
	height?: string | number;
	label?: string;
	leftIcon?: JSX.Element;
	rightIcon?: JSX.Element;
	rows?: number;
	width?: string | number;
};
