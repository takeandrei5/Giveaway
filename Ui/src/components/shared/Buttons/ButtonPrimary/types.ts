import { ButtonProps } from '@chakra-ui/react';

export type ButtonPrimaryProps = ButtonProps;
