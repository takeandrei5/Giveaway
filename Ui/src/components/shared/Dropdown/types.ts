export type DropdownProps = {
	options: DropdownOption[];
	onChangeHandler: (value: string) => void;
};

export type DropdownOption = {
	value: string | number;
	displayValue: string;
};
