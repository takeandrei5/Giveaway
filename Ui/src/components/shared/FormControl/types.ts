export type FormControlProps = {
	children: React.ReactElement;
	label: string;
	id: string;
	name: string;
};

export type useCheckFormIsInvalidProps = {
	isInvalid: boolean;
};
