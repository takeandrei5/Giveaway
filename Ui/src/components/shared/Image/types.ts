import { ImageProps as NextImageProps } from 'next/image';

export type ImageProps = NextImageProps & {
	height: string;
	width: string;
	backgroundColor?: string;
	borderRadius?: string;
	draggable?: boolean;
};
