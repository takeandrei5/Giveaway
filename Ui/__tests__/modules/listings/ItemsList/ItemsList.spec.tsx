import { ItemsList } from '@modules/listings/ItemsList';
import { ItemsListProps } from '@modules/listings/ItemsList/types';
import { render } from '@testing-library/react';

jest.mock('dateformat', () => jest.fn());

describe('ItemsList', () => {
	const props: ItemsListProps = {
		items: [
			{
				id: '1',
				title: 'Item 1',
				image: 'https://via.placeholder.com/150',
				createdAt: new Date(),
			},
			{
				id: '2',
				title: 'Item 2',
				image: 'https://via.placeholder.com/150',
				createdAt: new Date(),
			},
			{
				id: '3',
				title: 'Item 3',
				image: 'https://via.placeholder.com/150',
				createdAt: new Date(),
			},
		],
	};

	it('should match snapshot', () => {
		// Act
		const component = render(<ItemsList {...props} />);

		// Assert
		expect(component).toBeTruthy;
	});
});
