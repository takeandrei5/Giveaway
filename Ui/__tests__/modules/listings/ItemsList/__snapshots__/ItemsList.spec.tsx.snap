// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList should match snapshot 1`] = `
<div>
  <div
    class="css-ugh95v"
  >
    <div
      class="css-16nnvdh"
      data-testid="item"
    >
      <div
        class="css-qu5wq6"
      >
        <div
          class="css-n580hg"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
            />
            <img
              alt="Item 1"
              data-nimg="responsive"
              decoding="async"
              draggable="false"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="css-c5zn1r"
        >
          <span
            class="chakra-text css-inmt5s"
            data-testid="typography"
          >
            Item 1
          </span>
          <div
            class="css-8xl60i"
          >
            <span
              class="chakra-text css-1ysmetk"
              data-testid="typography"
            >
              Added on: 
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-16nnvdh"
      data-testid="item"
    >
      <div
        class="css-qu5wq6"
      >
        <div
          class="css-n580hg"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
            />
            <img
              alt="Item 2"
              data-nimg="responsive"
              decoding="async"
              draggable="false"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="css-c5zn1r"
        >
          <span
            class="chakra-text css-inmt5s"
            data-testid="typography"
          >
            Item 2
          </span>
          <div
            class="css-8xl60i"
          >
            <span
              class="chakra-text css-1ysmetk"
              data-testid="typography"
            >
              Added on: 
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-16nnvdh"
      data-testid="item"
    >
      <div
        class="css-qu5wq6"
      >
        <div
          class="css-n580hg"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
            />
            <img
              alt="Item 3"
              data-nimg="responsive"
              decoding="async"
              draggable="false"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="css-c5zn1r"
        >
          <span
            class="chakra-text css-inmt5s"
            data-testid="typography"
          >
            Item 3
          </span>
          <div
            class="css-8xl60i"
          >
            <span
              class="chakra-text css-1ysmetk"
              data-testid="typography"
            >
              Added on: 
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
